services/core/shared/indexdb/mysql.js

```diff
@@ -0,0 +1,183 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { Logger } = require('lisk-service-framework');
+const config = require('../../config');
+
+const logger = Logger();
+
+const connectionPool = {};
+const tablePool = {};
+
+const loadSchema = async (knex, tableName, tableConfig) => {
+	const { primaryKey, schema, indexes } = tableConfig;
+
+	if (await knex.schema.hasTable(tableName)) return knex;
+
+	await knex.schema
+	.createTable(tableName, table => {
+		Object.keys(schema).map(p => {
+			const kProp = (table[schema[p].type])(p);
+			if (schema[p].null === false) kProp.notNullable();
+			if (p === primaryKey) kProp.primary();
+			if (indexes[p]) kProp.index();
+			return kProp;
```

Comment:
As a TODO, please consider adding support for composite primary keys and foreign keys in the future.

```suggestion
			const kProp = (table[schema[p].type])(p);
			if (schema[p].null === false) kProp.notNullable();
			if (p === primaryKey) kProp.primary();
			if (indexes[p]) kProp.index();
			
			// TODO: Add support for composite primary keys and foreign keys
			return kProp;
```